discord/embeds.py

```diff
@@ -123,20 +65,32 @@ class EmbedAuthor(EmbedProxy):
     def __init__(
         self,
         name: str,
-        url: MaybeEmpty[str] = EmptyEmbed,
-        icon_url: MaybeEmpty[str] = EmptyEmbed,
-        proxy_icon_url: MaybeEmpty[str] = EmptyEmbed,
+        url: str | None = None,
+        icon_url: str | None = None,
     ) -> None:
-        layer = {
-            k: v
-            for k, v in locals().items()
-            if k in {"name", "url", "icon_url", "proxy_icon_url"}
-            and v is not EmptyEmbed
+        self.name: str = name
+        self.url: str | None = url
+        self.icon_url: str | None = icon_url
+        self.proxy_icon_url: str | None = None
+
+    @classmethod
+    def from_dict(cls, data: dict[str, str | None]) -> EmbedAuthor:
+        self = cls.__new__(cls)
+        self.name = data.get("name")
+        self.url = data.get("url")
+        self.icon_url = data.get("icon_url")
+        self.proxy_icon_url = data.get("proxy_icon_url")
+        return self
+
+    def to_dict(self) -> dict[str, str | None]:
+        return {
+            "name": str(self.name),
+            "url": str(self.url),
+            "icon_url": str(self.icon_url),
```

Comment:
The string conversion here is unnecessary and would also lead to a literal `"None"` being returned in urls.
```suggestion
            "name": self.name,
            "url": self.url,
            "icon_url": self.icon_url,
```

Reply:
Wouldn't removing the string conversion for `name` cause issues if anything else like an interger is passed?

Reply:
ppl shouldn't submit ints. their fault